src/plutus/cek.md

```diff
@@ -0,0 +1,136 @@
+                | 〈delay 𝑀 𝜌〉      // A delayed computation, with an
+                                   // associated environment
+                | 〈lam 𝑥 𝑀 𝜌〉      // A lambda abstraction, with an
+                                   // associated environment
+                | 〈constr 𝑖 𝑉*〉   // A constructor application, where
+                                   // all arguments are values
+                | 〈builtin 𝑏 𝑉* 𝜂〉 // A builtin application with all supplied
+                                   // arguments as values, and expecting
+                                   // at least one more argument
+
+𝜌 ∈ Environment ::= []        // An empty environment
+                  | 𝜌[𝑥 ↦ 𝑉]  // Associate 𝑥 with 𝑉 in the environment
+
+𝜂 ∈ Expected builtin arguments ::= [𝜄]  // One argument
+                                 | 𝜄⋅𝜂  // Two or more arguments
+
+𝑓 ∈ Frame ::= (force _)    // Awaiting a delayed computation to be forced
+            | [_ (𝑀, 𝜌)]  // An application awaiting the function, where the
+                          // argument is a term associated with an environment
+            | [_ 𝑉]       // An application awaiting the function, where the
+                          // argument is a value
+            | [𝑉 _]       // An application awaiting the argument, where the
+                          // function is a value
+            | (constr 𝑖 𝑉* _ (𝑀*, 𝜌))  // A constructor application awaiting
+                                       // an argument. The arguments before
+                                       // are values, and the arguments after
+                                       // are terms to be evaluated.
+            | (case _ (𝑀*, 𝜌))        // A case expression awaiting the scrutinee
+```
+
+The CEK machine has two main kinds of states:
+- `𝑠; 𝜌 ⊳ 𝑀` denotes evaluating term `𝑀` with environment `𝜌` and stack `𝑠`.
+- `𝑠 ⊲ 𝑉` denotes returning a value `𝑉` to stack `𝑠`.
+
+A value is a fully evaluated term, and an environment is a map binding variables to values.
+A stack frame contains a hole to represent a pending value, and the context needed to continue evaluation once the value is received.
+
+To evaluate a Plutus program containing a term `𝑀`, the machine starts from state `[]; [] ⊳ 𝑀`, and based on the following transition table, proceeds as follows:
+- If the current CEK machine state matches the From State, and the associated condition (if exists) is met, then the CEK machine transitions into the To State.
+- If the machine arrives at state `◻𝑉`, the machine terminates with success, yielding `𝑉` as final result.
+- If the machine gets stuck (i.e., no rule applies) or arrives at state `⬥`, the evaluation terminates with a failure.
+
+|Rule|From State        |  To State        | Condition |
+|:--:|:-----------------|:-----------------|:---------------|
+| 1  | `𝑠; 𝜌 ⊳ 𝑥`   | `𝑠 ⊲ 𝜌[𝑥]` | 𝑥 is bound in 𝜌 |
+| 2  | `𝑠; 𝜌 ⊳ (con T 𝑐)`   | `𝑠 ⊲ 〈con T 𝑐〉` |  |
```

Comment:
It would be nice to explain the subtle difference between `(con T 𝑐)` and `〈con T 𝑐〉`

Reply:
It should be clear from the definition that one is a term and the other is a value (the definition of `(con T 𝑐)` is not here yet, will add to a separate page).